ansible_runner/interface.py

```diff
@@ -91,7 +93,15 @@ def init_runner(**kwargs):
     event_callback_handler = kwargs.pop('event_handler', None)
     status_callback_handler = kwargs.pop('status_handler', None)
     artifacts_handler = kwargs.pop('artifacts_handler', None)
-    cancel_callback = kwargs.pop('cancel_callback', None)
+    if kwargs.get('cancel_callback'):
+        cancel_callback = kwargs.pop('cancel_callback')
+    # Only the main thread is allowed to set a new signal handler
+    elif threading.current_thread() is threading.main_thread():
```

Comment:
Runner has some tests that will call runner inside of a thread. This exposed the issue of trying to set a SignalHandler when runner isn't run inside the main thread.

This seems like it would be an issue when runner is called asynchronously (`ansible_runner.interface.run_async`). This has not been an issue for the awx use case where receptor calls runner via `ansible-runner worker`.

Reply:
✅